test/terra/backends/qasm_simulator/qasm_unitary_gate.py

```diff
@@ -54,6 +57,29 @@ def test_random_unitary_gate(self):
         self.assertSuccess(result)
         self.compare_counts(result, circuits, targets, delta=0.05 * shots)
 
+    def test_random_unitary_gate_with_permutations(self):
+        """Test simulation with random unitary gate with permutations."""
+        all_permutations = list(itertools.permutations([0, 1, 2]))
+        unitary_matrix = random_unitary(8, seed=8)
+        n = 3
+        shots = 2000
+        qr = QuantumRegister(n, 'qr')
+        cr = ClassicalRegister(n, 'cr')
+        regs = (qr, cr)
+    
+        for perm in all_permutations:
+            circuit = QuantumCircuit(*regs)
+            circuit.unitary(unitary_matrix, perm)
+            circuit.barrier(qr)
+            circuit.measure(qr, cr)
+            result = execute([circuit], self.SIMULATOR, shots=shots,
+                         **self.BACKEND_OPTS).result()
+            
+            state = Statevector.from_label(n * '0').evolve(unitary_matrix, perm)
+            counts = state.sample_counts(shots=shots)
+            hex_counts = {hex(int(key, 2)): val for key, val in counts.items()}
+            self.assertSuccess(result)
+            self.compare_counts(result, [circuit], [hex_counts], delta=0.05 * shots)
```

Comment:
Since there are only three qubits, it's better to remove the measurement from the circuit, and compare statevectors instead of counts. In this case note that your circuit should be;
```
circuit = QuantumCircuit(n)
```

Reply:
This doesn't work because the test runs on other simulation methods, include `density_matrix`. And `density_matrix` does not support extracting a statevector. So left it with measurements.

Reply:
See how it's done in `test/terra/backends/qasm_simulator/qasm_standard_gates.py`

Reply:
I prefer to keep it similar to the other tests in this class.